<launch>
    
    <!--==========================-->
    <!-- Arguments and Parameters -->
    <!--==========================-->
    <arg name="manual_deadman" default="4"/>
    <arg name="autonomous_deadman" default="5"/>
    <arg name="timeout" default="1.0"/>
    <arg name="joy_dev" default="/dev/input/js0"/>
    <arg name="joy_clamp_scale" default="1.0"/>
    <arg name="max_velocity" default="0.25"/>
    <!-- options: velocity, pmw -->
    <arg name="pedal_mode" default="velocity"/>
    <!-- options: relative, absolute -->
    <arg name="steering_mode" default="relative"/>

    <!--========================-->
    <!-- Joystick Control Nodes -->
    <!--========================--> 
    <node pkg="joy" type="joy_node" name="joy_node">
        <param name="dev" value="$(arg joy_dev)" />
        <param name="deadzone" value="0.3" />
        <param name="autorepeat_rate" value="20" />
    </node>

    <node pkg="motion_testing" type="joystick_control.py" name="joystick_control" output="screen">
        <param name="manual_deadman" value="$(arg manual_deadman)"/>
        <param name="autonomous_deadman" value="$(arg autonomous_deadman)"/>
        <param name="timeout" value="$(arg timeout)"/>
        <param name="vel_max" value="$(arg max_velocity)"/>
        <param name="pwm_max" value="255"/>
        <param name="pedal_mode" value="$(arg pedal_mode)"/>
        <param name="steering_mode" value="$(arg steering_mode)"/>
        <param name="clamp_scale" value="$(arg joy_clamp_scale)"/>
    </node>
    
</launch>
